reconcile/dynatrace_token_provider/integration.py

```diff
@@ -231,13 +232,34 @@ def reconcile(self, dry_run: bool, dependencies: Dependencies) -> None:
                                 ocm_env_name=ocm_env_name,
                             )
                     except Exception as e:
-                        unhandled_exceptions.append(
-                            f"{ocm_env_name}/{cluster.organization_id}/{cluster.external_id}: {e}"
-                        )
+                        unhandled_exceptions.append((
+                            e,
+                            f"{ocm_env_name}/{cluster.organization_id}/{cluster.external_id}: {e}",
+                        ))
                 self._expose_token_metrics()
 
-        if unhandled_exceptions:
-            raise ReconcileErrorSummary(unhandled_exceptions)
+        unexpected_errors_cnt = 0
+        for err, _ in unhandled_exceptions:
+            if (
+                isinstance(err, OCMAPIError)
+                and err.status_code >= 500
+                and (
+                    "api.integration.openshift.com" in err.endpoint
+                    or "api.stage.openshift.com" in err.endpoint
+                )
+            ):
+                # OCMAPI in int/stage can fail. Lets for now ignore those
```

Comment:
what would it change if we ignored 5xx on prod as well? Do we really need to fail or skipping is fine as well?

Reply:
Good point. 5xx on prod would most likely indicate a syncset/manifest API issue on prod which is pretty serious - so I thought for now it would be fine to keep old behavior. However, since we will expose those as separate metrics anyways, we can have more fine-grained alerts and severities configured outside of this code and dont need to block.

Will change this to skip on all 5xx on all OCM envs.